retval=DENY msg="sorry, that domain isn't in my list of allowed rcpthosts (#5.7.1)"

### Expected behavior
Deliver to local.domain by qmail:25 local interface.
Deliver to internet.domain. via default gw.


### Observed behavior
2016-11-14T14:58:48.848Z [INFO] [C49D1EBB-5135-4716-8E50-5551275DE12B.1] [rcpt_to.qmail_deliverable] qmail-command in dot-qmail
2016-11-14T14:58:48.850Z [NOTICE] [C49D1EBB-5135-4716-8E50-5551275DE12B.1] [core] sender <helpdesk@loacal.domain> code=CONT msg=""
2016-11-14T14:58:48.853Z [INFO] [C49D1EBB-5135-4716-8E50-5551275DE12B.1] [rcpt_to.qmail_deliverable] not local
2016-11-14T14:58:48.853Z [INFO] [C49D1EBB-5135-4716-8E50-5551275DE12B.1] [core] hook=rcpt plugin=rcpt_to.qmail_deliverable function=hook_rcpt params="<zgloszenia.abcd@internet.domain>" retval=OK msg=""
2016-11-14T14:58:48.854Z [NOTICE] [C49D1EBB-5135-4716-8E50-5551275DE12B.1] [core] recipient <zgloszenia.abcd@internet.domain> code=OK msg="" sender="helpdesk@loacal.domain"
2016-11-14T14:58:48.963Z [INFO] [C49D1EBB-5135-4716-8E50-5551275DE12B.1] [clamd] pass:clean
2016-11-14T14:58:48.971Z [INFO] [C49D1EBB-5135-4716-8E50-5551275DE12B.1] [avg] time=7ms code=200 response="ok"
2016-11-14T14:58:48.972Z [NOTICE] [C49D1EBB-5135-4716-8E50-5551275DE12B.1] [core] message mid="<ERP@loacal.domain>" size=1021 rcpts=1/0/0 delay=0.016 code=CONT msg=""
2016-11-14T14:58:48.973Z [ERROR] [C49D1EBB-5135-4716-8E50-5551275DE12B.1] [dkim_sign] skipped: missing dkim private key
2016-11-14T14:58:48.973Z [INFO] [C49D1EBB-5135-4716-8E50-5551275DE12B.1] [queue/smtp_forward] forwarding to 127.0.0.8:25
2016-11-14T14:58:48.974Z [INFO] [C49D1EBB-5135-4716-8E50-5551275DE12B.1] [core] hook=queue_outbound plugin=queue/smtp_forward function=hook_queue params="" retval=DENY msg="sorry, that domain isn't in my list of allowed rcpthosts (#5.7.1)"
2016-11-14T14:58:48.975Z [INFO] [C49D1EBB-5135-4716-8E50-5551275DE12B.1] [core] hook=deny plugin=karma function=hook_deny params="902" retval=OK msg=""
2016-11-14T14:58:48.975Z [INFO] [C49D1EBB-5135-4716-8E50-5551275DE12B.1] [core] deny(soft?) overriden by deny hook
2016-11-14T14:58:48.975Z [NOTICE] [C49D1EBB-5135-4716-8E50-5551275DE12B.1] [core] queue code=CONT msg=" (C49D1EBB-5135-4716-8E50-5551275DE12B.1)"
2016-11-14T14:58:48.980Z [NOTICE] [C49D1EBB-5135-4716-8E50-5551275DE12B.1] [core] queue code=OK msg="Message Queued"
2016-11-14T14:58:48.980Z [INFO] [C49D1EBB-5135-4716-8E50-5551275DE12B.1] [karma] score: 8, good: 8012, bad: 2178, connections: 10259, history: 5834, awards: 163, pass:relaying, fail:deny:queue/smtp_forward
2016-11-14T14:58:48.982Z [INFO] [C49D1EBB-5135-4716-8E50-5551275DE12B.1] [karma] score: 8, good: 8012, bad: 2178, connections: 10259, history: 5834, awards: 163, pass:relaying, fail:deny:queue/smtp_forward
2016-11-14T14:58:48.983Z [NOTICE] [C49D1EBB-5135-4716-8E50-5551275DE12B.1] [core] disconnect ip=10.60.11.61 rdns="DNSERROR" helo="loacal.domain" relay=Y early=N esmtp=Y tls=Y pipe=N errors=0 txns=1 rcpts=1/0/0 msgs=1/0/0 bytes=1021 lr="" time=0.174
2016-11-14T14:58:48.984Z [INFO] [C49D1EBB-5135-4716-8E50-5551275DE12B.1.1] [outbound] Looking up A records for: mx2.internet.domain
2016-11-14T14:58:48.985Z [INFO] [C49D1EBB-5135-4716-8E50-5551275DE12B.1.1] [outbound] Attempting to deliver to: 91.242.220.133:25 (0) (5)
2016-11-14T14:58:49.469Z [NOTICE] [C49D1EBB-5135-4716-8E50-5551275DE12B.1.1] [outbound] delivered file=1479135528976_0_36131_1198.haraka domain="internet.domain" host="mx2.internet.domain" ip=91.242.220.133 port=25 mode=SMTP tls=N auth=N response="" delay=0.493 fails=0 rcpts=1/0/0

### Steps to reproduce
Send one email to:
zgloszenia.abcd@internet.domain
ERP@local.domain
local.domain is serving by qmail backend. Standart Mail-Toaster installation.
Why messages to local.domain are delivered via outside interface ?
Host is serving local.domain - it's true internet domain. I changed it only for example.

--- a/.travis.yml
+++ b/.travis.yml
@@ -34,7 +34,7 @@ script:
    - sh -n *.sh
    - shellcheck -e SC2009,SC2039,SC2086,SC2153,SC2016,SC1004,SC2119 *.sh
    - sh -n qmail/run.sh
-   - shellcheck qmail/run.sh
+   - shellcheck -e SC2039 qmail/run.sh
 
 # after_success:
 # after_failure:

--- a/.travis.yml
+++ b/.travis.yml
@@ -33,6 +33,8 @@ before_script:
 script:
    - sh -n *.sh
    - shellcheck -e SC2009,SC2039,SC2086,SC2153,SC2016,SC1004,SC2119 *.sh
+   - sh -n qmail/run.sh
+   - shellcheck qmail/run.sh
 
 # after_success:
 # after_failure:

--- a/provision-vpopmail.sh
+++ b/provision-vpopmail.sh
@@ -228,7 +228,7 @@ configure_qmail()
 configure_vpopmail()
 {
 	tell_status "setting up daemon supervision"
-	fetch -o - "$TOASTER_SRC_URL/qmail/run.txt" | stage_exec sh
+	fetch -o - "$TOASTER_SRC_URL/qmail/run.sh" | stage_exec sh
 
 	if [ ! -d "$ZFS_DATA_MNT/vpopmail/domains/$TOASTER_MAIL_DOMAIN" ]; then
 		tell_status "ATTN: Your postmaster password is..."

--- a/qmail/run.sh
+++ b/qmail/run.sh
@@ -257,7 +257,7 @@ EO_QMAILCTL
 install_vpopmail_etc()
 {
 	ETC="/usr/local/vpopmail/etc"
-	if [ -z "$ETC/tcp.smtp" ]; then
+	if [ -s "$ETC/tcp.smtp" ]; then
 		echo -n "Re"
 	fi
 
@@ -327,8 +327,7 @@ install_vpopmaild_run()
 
 	echo "installing $RUN"
 	mkdir -p "$SUP/vpopmaild/log/main"
-	#tee "$RUN" <<'EO_VPOPMAILD'
-	cat <<'EO_VPOPMAILD' > $RUN
+	tee "$RUN" <<'EO_VPOPMAILD'
 #!/bin/sh
 PATH=/var/qmail/bin:/usr/local/bin:/usr/bin:/bin
 export PATH
@@ -337,7 +336,7 @@ EO_VPOPMAILD
 	chmod 755 $RUN
 
 	echo "installing $LOGRUN"
-	cat <<'EO_VPOPMAILD_LOG' > "$LOGRUN"
+	tee "$LOGRUN" <<'EO_VPOPMAILD_LOG'
 #!/bin/sh
 PATH=/var/qmail/bin:/usr/local/bin:/usr/bin:/bin
 export PATH
@@ -375,8 +374,7 @@ exec /usr/local/bin/setuidgid qmaill /usr/local/bin/multilog ./main
 EO_DELIVERABLED_RUN
 	chmod 755 "$LOGRUN"
 
-	if [ ! `pkg query %n -F p5-HTTP-Daemon` ];
-	then
+	if [ ! "$(pkg query %n -F p5-HTTP-Daemon)" ]; then
 		echo "Installing HTTP::Daemon"
 		pkg install -y p5-HTTP-Daemon
 		make -C /usr/ports/www/p5-HTTP-Daemon deinstall install clean
@@ -402,9 +400,9 @@ install_qmail_chkuser()
 		exit
 	fi
 
-	cd $PORTBUILDDIR
+	cd $PORTBUILDDIR || exit
 	fetch -o - $CHKPATCH | tar -xzOf - | patch -p1
-	if [ -f *.rej ]; then
+	if stat -t ./*.rej >/dev/null 2>&1; then
 		echo "Patch did not apply cleanly, I refuse to proceed!"
 		exit
 	fi
@@ -453,9 +451,8 @@ chmod 755 /service/*/run
 chmod 755 /service/*/log/run
 chown -R qmaill $SUP/*/log
 
-if [ ! `grep svscan_enable /etc/rc.conf` ];
-then
-	echo 'svscan_enable="YES"' >> /etc/rc.conf
+if ! grep -qs svscan_enable /etc/rc.conf; then
+	sysrc svscan_enable=YES
 fi
 
 service svscan restart

--- a/qmail/run.sh
+++ b/qmail/run.sh
@@ -257,68 +257,61 @@ EO_QMAILCTL
 install_vpopmail_etc()
 {
 	ETC="/usr/local/vpopmail/etc"
-	if [ -z "$ETC/tcp.smtp" ];
-	then
+	if [ -z "$ETC/tcp.smtp" ]; then
 		echo -n "Re"
 	fi
 
 	echo "installing $ETC/tcp.smtp"
 	mkdir -p $ETC || exit
-	#tee $ETC/tcp.smtp <<'EO_VPOPMAIL_ETC'
-	cat <<'EO_VPOPMAIL_ETC' > $ETC/tcp.smtp
+	tee "$ETC/tcp.smtp" <<EO_VPOPMAIL_ETC
 # if the chkuser patch is compiled into qmail,
 # CHKUSER_MBXQUOTA rejects messages when the users mailbox quota is filled
 127.0.0.1:allow,RELAYCLIENT=""
-172.16.15.9:allow,RELAYCLIENT=""
+${JAIL_NET_PREFIX}.9:allow,RELAYCLIENT=""
 :allow,CHKUSER_MBXQUOTA="99"
 EO_VPOPMAIL_ETC
+
 	/usr/local/bin/qmailctl cdb
     if [ -f "$ETC/tcp.smtp.cdb" ]; then
         if [ -f "$SUP/qmail-smtpd/run" ]; then
             echo "adding tcp.smtp.cdb to qmail-smtpd/run"
-            sed -i .bak -e '/-u 89/ s/-g 82/-g 82 -x \/usr\/local\/vpopmail\/etc\/tcp.smtp.cdb/' "$SUP/qmail-smtpd/run"
+            sed -i .bak \
+                -e '/-u 89/ s/-g 82/-g 82 -x \/usr\/local\/vpopmail\/etc\/tcp.smtp.cdb/' \
+                "$SUP/qmail-smtpd/run"
         fi
     fi
 }
 
 install_symlinks()
 {
-	if [ ! -L "/service" ];
-	then
+	if [ ! -L "/service" ]; then
 		echo "ln -s /var/service /service"
 		ln -s /var/service /service
 	fi
 
-	if [ ! -L "/service/qmail-smtpd" ];
-	then
+	if [ ! -L "/service/qmail-smtpd" ]; then
 		echo "supervising qmail-smtpd"
 		ln -s /var/qmail/supervise/qmail-smtpd /service/qmail-smtpd
 	fi
 
-	if [ ! -L "/service/qmail-send" ];
-	then
+	if [ ! -L "/service/qmail-send" ]; then
 		echo "supervising qmail-send"
 		ln -s /var/qmail/supervise/qmail-send /service/qmail-send
 	fi
 
-	if [ ! -L "/service/vpopmaild" ];
-	then
+	if [ ! -L "/service/vpopmaild" ]; then
 		echo "supervising vpopmaild"
 		ln -s /var/qmail/supervise/vpopmaild /service/vpopmaild
 	fi
 
-	if [ ! -L "/service/qmail-deliverabled" ];
-	then
+	if [ ! -L "/service/qmail-deliverabled" ]; then
 		echo "supervising qmail-deliverabled"
 		ln -s /var/qmail/supervise/deliverabled /service/qmail-deliverabled
 	fi
 
-	if [ ! -L "/service/clear" ];
-	then
-		if [ -d "/var/qmail/supervise/clear" ];
-		then
-# TODO: test this
-			#ln -s /var/qmail/supervise/clear /service/clear
+	if [ ! -L "/service/clear" ]; then
+		if [ -d "/var/qmail/supervise/clear" ]; then
+			ln -s /var/qmail/supervise/clear /service/clear
 		fi
 	fi
 }
@@ -427,8 +420,7 @@ install_qmail_chkuser()
 install_clear_run()
 {
 	RUN="$SUP/clear/run"
-	if [ -x "$RUN" ];
-	then
+	if [ -x "$RUN" ]; then
 		echo -n "Re"
 	fi
 

